File: src/class13/deliverableexercise/hash/java/TitularTeacher.java
package class13.deliverableexercise.hash.java;

public class TitularTeacher extends Teacher {
    private String specialty;

    public TitularTeacher(String name, String lastName, Integer houseTime, Integer teacherCode, String specialty) {
        super(name, lastName, houseTime, teacherCode);
        this.specialty = specialty;
    }

    public String getSpecialty() {
        return specialty;
    }

    public void setSpecialty(String specialty) {
        this.specialty = specialty;
    }
}

File: app/src/main/java/intellij/class13/deliverableexercise/hash/kotlin/TitularTeacher.kt
package intellij.class13.deliverableexercise.hash.kotlin

import class13.deliverableexercise.hash.kotlin.Teacher

class TitularTeacher(
    name: String,
    lastName: String,
    houseTime: Int,
    teacherCode: Int,
    var specialty: String
) :
    Teacher(name, lastName, houseTime, teacherCode)